llvm/include/llvm/Transforms/Utils/ValueMapper.h

```diff
@@ -175,6 +179,8 @@ class ValueMapper {
   Constant *mapConstant(const Constant &C);
 
   void remapInstruction(Instruction &I);
+  void remapDPValue(Module *M, DPValue &V);
+  void remapDPValueRange(Module *M, iterator_range<simple_ilist<DPValue>::iterator> Range);
```

Comment:
It's taking an iterator range, but we're naming it in a weird way -- to avoid including DebugProgramInstruction.h to get the full definition of DPValue we're pre-declaring DPValue and then naming what a list-iterator type for it would be, rather than  using DPValue::self_iterator. Then wrapping that in iterator_range.

It looks like there's been a decent effort to keep Instruction.h and friends out of this file, so I'd shy away from adding more #includes. Perhaps we can live with a using/typedef for the iterator type, `using DPValueIteratorT = simple_ilist<DPValue>::iterator`?

Reply:
That makes sense, thanks. It's fine to leave as is IMO - I just misread the type slightly.

Reply:
GitHub reviews continue to throw me - I hadn't seen that you've already made that change. The new change also looks good, and is less confusing at a glance.